consensus/src/creation/mod.rs

```diff
@@ -69,7 +69,7 @@ async fn create_unit<H: Hasher>(
             },
             _ = exit => {
                 info!(target: "AlephBFT-creator", "Received exit signal.");
-                return Err(());
+                return Ok(None);
```

Comment:
Oh no, this is definitely bad. If you really need to disambiguate the cases here, you should introduce another error type, but I suspect rebuilding the system to avoid deadlocks with `Terminator` as @fixxxedpoint suggested should be the better approach.